src/platform/Linux/bluez/Helper.cpp

```diff
@@ -448,15 +445,15 @@ static gboolean BluezCharacteristicAcquireWrite(BluezGattCharacteristic1 * aChar
 #endif // CHIP_ERROR_LOGGING
         ChipLogError(DeviceLayer, "FAIL: socketpair: %s in %s", StringOrNullMarker(errStr), __func__);
         g_dbus_method_invocation_return_dbus_error(aInvocation, "org.bluez.Error.Failed", "FD creation failed");
-        goto exit;
+        return FALSE;
     }
 
-    options    = g_variant_dict_new(aOptions);
-    option_mtu = g_variant_dict_lookup_value(options, "mtu", G_VARIANT_TYPE_UINT16);
-    VerifyOrExit(
-        option_mtu != nullptr, ChipLogError(DeviceLayer, "FAIL: No MTU in options in %s", __func__);
+    options    = GAutoPtr<GVariantDict>(g_variant_dict_new(aOptions));
+    option_mtu = GAutoPtr<GVariant>(g_variant_dict_lookup_value(options.get(), "mtu", G_VARIANT_TYPE_UINT16));
```

Comment:
If only one key is retrieved from the dictionary, one can use `g_variant_lookup()` or `g_variant_lookup_value()` without dict prior creation. It could simplify the code a bit. Please look for other `g_variant_dict_lookup_value()` usages.